pkg/controller/expcertificates/issuing/issuing_controller.go

```diff
@@ -270,6 +270,11 @@ func (c *controller) ProcessItem(ctx context.Context, key string) error {
 	// If the CertificateRequest is valid and ready, verify its status and issue
 	// accordingly.
 	if cond.Reason == cmapi.CertificateRequestReasonIssued {
+		// Translate to correct KeyEncoding as specified by Certificate resource
+		pkData, err = utilpki.TranslateToKeyEncoding(pkData, crt.Spec.KeyEncoding)
```

Comment:
Instead of adding this method here (which goes from `[]byte` to `[]byte`), can we instead adjust the `issueCertificate` method to accept a `crypto.PrivateKey` for the 'privateKey' argument (instead of a `[]byte`), so that here we can pass `pk` instead of `pkData`.

You can then make use of the existing method 'EncodePrivateKey' instead which goes from `crypto.PrivateKey` to a `[]byte`: https://github.com/jetstack/cert-manager/blob/8a69edb649adad94d2c2abf33e1baaa1ca43839a/pkg/util/pki/generate.go#L111-L130

```
// EncodePrivateKey will encode a given crypto.PrivateKey by first inspecting
// the type of key encoding and then inspecting the type of key provided.
// It only supports encoding RSA or ECDSA keys.
func EncodePrivateKey(pk crypto.PrivateKey, keyEncoding v1alpha2.KeyEncoding) ([]byte, error) {
```

Reply:
Aaah ok.. I got confused by interfaces in Go